Flaky test: HotReloadIT port-addressed tests fail with 'Address already in use'

## Description

Two tests in `HotReloadIT` are flaky and intermittently fail in CI with port binding errors.

## Failure Details

**Test class**: `io.kroxylicious.it.HotReloadIT`

**Failing tests**:
1. `shouldSupportPortReuseAcrossReconfigures` - fails at startup with `LifecycleException: Startup completed exceptionally`
2. `shouldSurfaceBindFailureAsReconfigureErrorWithoutBlockingOtherAdds` - fails when attempting to open a socket with `UncheckedIOException: java.net.BindException: Address already in use`

**Error summary**:
```
[ERROR] Tests run: 8, Failures: 0, Errors: 2, Skipped: 0, Time elapsed: 30.55 s <<< FAILURE! -- in io.kroxylicious.it.HotReloadIT
[ERROR]   HotReloadIT.shouldSupportPortReuseAcrossReconfigures:366 » Lifecycle Startup completed exceptionally
[ERROR]   HotReloadIT.shouldSurfaceBindFailureAsReconfigureErrorWithoutBlockingOtherAdds:317->openSocketOnPort:496 UncheckedIO java.net.BindException: Address already in use
```

## CI Run

Failed in workflow run: https://github.com/kroxylicious/kroxylicious/actions/runs/26933746138/job/79461670206?pr=4064

--- a/kroxylicious-integration-tests/src/test/java/io/kroxylicious/it/HotReloadIT.java
+++ b/kroxylicious-integration-tests/src/test/java/io/kroxylicious/it/HotReloadIT.java
@@ -466,6 +466,7 @@ class HotReloadIT extends BaseIT {
                 .pollInterval(Duration.ofMillis(100))
                 .untilAsserted(() -> {
                     try (var s = new ServerSocket()) {
+                        s.setReuseAddress(true);
                         s.bind(new InetSocketAddress((InetAddress) null, port));
                     }
                 });
@@ -479,6 +480,7 @@ class HotReloadIT extends BaseIT {
         ServerSocket s = null;
         try {
             s = new ServerSocket();
+            s.setReuseAddress(true);
             s.bind(new InetSocketAddress((InetAddress) null, port));
             return s;
         }

